Tribler/Test/Community/Multichain/test_multichain_database.py

```diff
@@ -195,3 +196,87 @@ def test_block_to_dictionary(self):
+        """
+        self.block2.total_up = 0
+
+        self.db.add_block(self.block1)
+        self.db.add_block(self.block2)
+
+        self.assertEqual(self.block1.total_up, self.db.total_up(self.block1.public_key))
+        self.assertEqual(0, self.db.total_up(self.block2.public_key))
+        self.assertEqual(0, self.db.total_up(self.block3.public_key))
+
+    @blocking_call_on_reactor_thread
+    def test_total_down(self):
+        """
+        The database should return the correct amount of downloaded data.
+        """
+        self.block2.total_down = 0
+
+        self.db.add_block(self.block1)
+        self.db.add_block(self.block2)
+
+        self.assertEqual(self.block2.total_down, self.db.total_down(self.block2.public_key))
+        self.assertEqual(0, self.db.total_down(self.block2.public_key))
+        self.assertEqual(0, self.db.total_down(self.block3.public_key))
+
+    @blocking_call_on_reactor_thread
+    def test_neighbors(self):
+        """
+        The database should return the correct list of neighbors and the traffic to and from them.
+        """
+        focus_block1 = TestBlock()
+        focus_block2 = TestBlock()
+
+        # All blocks have the same public key
+        self.block2.public_key = self.block1.public_key
+        self.block3.public_key = self.block1.public_key
+
+        self.block1.link_public_key = focus_block1.public_key
+        self.block2.link_public_key = focus_block1.public_key
+        self.block3.link_public_key = focus_block2.public_key
+
+        # Add all blocks + one redundant block
+        self.db.add_block(self.block1)
+        self.db.add_block(self.block2)
+        self.db.add_block(self.block3)
+        self.db.add_block(focus_block1)
+
+        expected_result = {focus_block1.public_key:
+                               {"up": self.block1.up + self.block2.up, "down": self.block1.down + self.block2.down},
+                           focus_block2.public_key: {"up": self.block3.up, "down": self.block3.down}}
+
+        self.assertDictEqual(expected_result, self.db.neighbor_list(self.block1.public_key))
+
+    @blocking_call_on_reactor_thread
+    def test_random_dummy_data(self):
+        """
+        The database should contain 104 rows when random dummy data is used.
+        """
+        self.db.use_dummy_data(use_random=True)
+
+        num_rows = self.db.execute(u"SELECT count (*) FROM multi_chain").fetchone()[0]
+        self.assertEqual(num_rows, 104)
+        self.assertTrue(self.db.dummy_setup)
+
+    @blocking_call_on_reactor_thread
+    def test_static_dummy_dta(self):
+        """
+        The database should contain the fixed dataset when non-random dummy data is used.
+        """
+        self.db.use_dummy_data(use_random=False)
+
+        num_rows = self.db.execute(u"SELECT count (*) FROM multi_chain").fetchone()[0]
+        self.assertEqual(num_rows, 56)
+        self.assertTrue(self.db.dummy_setup)
+
+        self.db.use_dummy_data(use_random=True)
```

Comment:
Why is this part of the test? It seems like unexpected behavior rather than expected behavior